iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -209,6 +244,307 @@ class GlobalStoreOpConversion : public OpConversionPattern<StoreOpTy> {
+        /*result=*/emitc::OpaqueType::get(ctx, "iree_vm_ref_t*"),
+        /*operand=*/refOp.getResult());
+
+    auto refTypeOp = rewriter.create<emitc::CallOp>(
+        /*location=*/loc,
+        /*type=*/emitc::OpaqueType::get(ctx, "iree_vm_ref_type_t"),
+        /*callee=*/rewriter.getStringAttr("iree_vm_list_type_id"),
+        /*args=*/ArrayAttr{},
+        /*templateArgs=*/ArrayAttr{},
+        /*operands=*/ArrayRef<Value>{});
+
+    rewriter.create<emitc::CallOp>(
+        /*location=*/loc,
+        /*type=*/TypeRange{},
+        /*callee=*/rewriter.getStringAttr("iree_vm_ref_wrap_assign"),
+        /*args=*/ArrayAttr{},
+        /*templateArgs=*/ArrayAttr{},
+        /*operands=*/
+        ArrayRef<Value>{listOp.getResult(), refTypeOp.getResult(0),
+                        refPtrOp.getResult()});
+
+    return success();
+  }
+};
+
+template <typename GetOpTy>
+class ListGetOpConversion : public OpConversionPattern<GetOpTy> {
+  using OpConversionPattern<GetOpTy>::OpConversionPattern;
+
+ private:
+  LogicalResult matchAndRewrite(
+      GetOpTy getOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto ctx = getOp.getContext();
+    auto loc = getOp.getLoc();
+
+    Optional<StringRef> valueTypeEnum;
+    Optional<StringRef> valueExtractor;
+
+    std::tie(valueTypeEnum, valueExtractor) =
+        TypeSwitch<Operation *,
+                   std::pair<Optional<StringRef>, Optional<StringRef>>>(
+            getOp.getOperation())
+            .Case<IREE::VM::ListGetI32Op>([&](auto op) {
+              return std::make_pair(StringRef("IREE_VM_VALUE_TYPE_I32"),
+                                    StringRef("vm_list_value_extract_i32"));
+            })
+            .template Case<IREE::VM::ListGetI64Op>([&](auto op) {
+              return std::make_pair(StringRef("IREE_VM_VALUE_TYPE_I64"),
+                                    StringRef("vm_list_value_extract_i64"));
+            })
+            .Default([](Operation *) { return std::make_pair(None, None); });
+
+    if (!valueTypeEnum.hasValue() || !valueExtractor.hasValue()) {
+      return getOp.emitOpError() << " not handeled";
+    }
+
+    auto valueOp = rewriter.create<emitc::ConstOp>(
+        /*location=*/loc,
+        /*resultType=*/emitc::OpaqueType::get(ctx, "iree_vm_value_t"),
+        /*value=*/StringAttr::get(ctx, ""));
+
+    auto valuePtrOp = rewriter.create<emitc::GetAddressOfOp>(
+        /*location=*/loc,
+        /*result=*/emitc::OpaqueType::get(ctx, "iree_vm_value_t*"),
+        /*operand=*/valueOp.getResult());
+
+    auto listDerefOp = rewriter.create<emitc::CallOp>(
+        /*location=*/loc,
+        /*type=*/emitc::OpaqueType::get(ctx, "iree_vm_list_t*"),
+        /*callee=*/rewriter.getStringAttr("iree_vm_list_deref"),
+        /*args=*/ArrayAttr{},
+        /*templateArgs=*/ArrayAttr{},
+        /*operands=*/ArrayRef<Value>{getOp.list()});
+
+    auto getValueOp = rewriter.create<emitc::CallOp>(
```

Comment:
also fails (as do likely others below) - the value will be undefined if it does so it's critical to check